src/kernels/PrecursorEigenSource.C

```diff
@@ -0,0 +1,87 @@
+#include "PrecursorEigenSource.h"
+
+registerMooseObject("MoltresApp", PrecursorEigenSource);
+
+InputParameters
+PrecursorEigenSource::validParams()
+{
+  InputParameters params = EigenKernel::validParams();
+  params += ScalarTransportBase::validParams();
+  params.addRequiredParam<unsigned int>("num_groups", "The total number of energy groups");
+  params.addRequiredCoupledVar("group_fluxes", "All the variables that hold the group fluxes. "
+                                               "These MUST be listed by decreasing "
+                                               "energy/increasing group number.");
+  params.addParam<unsigned int>("precursor_group_number",
+                                "What precursor group this kernel is acting on.");
+  params.addCoupledVar(
+      "temperature", 800, "The temperature used to interpolate material properties.");
+  params.addParam<Real>("prec_scale", 1, "Thefactor by which the neutron fluxes are scaled.");
+  return params;
+}
+
+PrecursorEigenSource::PrecursorEigenSource(const InputParameters & parameters)
+  : EigenKernel(parameters),
+    ScalarTransportBase(parameters),
+    _nsf(getMaterialProperty<std::vector<Real>>("nsf")),
+    _d_nsf_d_temp(getMaterialProperty<std::vector<Real>>("d_nsf_d_temp")),
+    _num_groups(getParam<unsigned int>("num_groups")),
+    _beta_eff(getMaterialProperty<std::vector<Real>>("beta_eff")),
+    _d_beta_eff_d_temp(getMaterialProperty<std::vector<Real>>("d_beta_eff_d_temp")),
+    _precursor_group(getParam<unsigned int>("precursor_group_number") - 1),
+    _temp(coupledValue("temperature")),
+    _temp_id(coupled("temperature")),
+    _prec_scale(getParam<Real>("prec_scale"))
```

Comment:
I haven't seen (or just don't remember) this kind of syntax used after a function C. What does it mean?

Reply:
This is the initialization list containing all references to be initialized. This is typical for Constructor functions in C++